pkg/gofr/container/datasources.go

```diff
@@ -395,3 +395,276 @@ type DgraphProvider interface {
 	Dgraph
 	provider
 }
+
+type OpenTSDBProvider interface {
+	OpenTSDB
+	provider
+}
+
+// OpenTSDB provides methods for GoFr applications to communicate with OpenTSDB
+// through its REST APIs. Each method corresponds to an API endpoint as defined in
+// the OpenTSDB documentation (http://opentsdb.net/docs/build/html/api_http/index.html#api-endpoints).
+type OpenTSDB interface {
+
+	// HealthChecker checks if the target OpenTSDB server is reachable.
+	// It returns an error if the server is unreachable, otherwise returns nil.
+	HealthChecker
+
+	// Put handles the 'POST /api/put' endpoint, allowing the storage of data in OpenTSDB.
+	//
+	// Parameters:
+	// - data: A slice of DataPoint objects, which must contain at least one instance.
+	// - queryParam: Can be one of the following:
+	//   - client.PutRespWithSummary: Requests a summary of the put operation.
+	//   - client.PutRespWithDetails: Requests detailed information about the put operation.
+	//   - An empty string (""): Indicates no additional response details are required.
+	//
+	// Return:
+	// - On success, it returns a pointer to a PutResponse, along with the HTTP status code and relevant response information.
+	// - On failure (due to invalid parameters, response parsing errors, or OpenTSDB connectivity issues), it returns an error.
+	//
+	// Notes:
+	// - Use 'PutRespWithSummary' to receive summarized information about the data that was stored.
+	// - Use 'PutRespWithDetails' for a more comprehensive breakdown of the put operation.
+	Put(ctx context.Context, data any, queryParam string) (any, error)
```

Comment:
Here and everywhere

I'm surprised by the type `any` and `any` reply

The documentation is pretty clear 

http://opentsdb.net/docs/build/html/api_http/put.html

Reply:
Usually one was to do check the interface is satisfied is to add something like this in code

```
var _ gofr.Request = YourObject{}
```

This way, you will get a typecheck error aside your implementation

Reply:
I am using any to prevent adding opentsdb dependency in the container package where I am defining the interface. I have changed the function definitions so that it works .....